common/testing/updateutils/update.go

```diff
@@ -0,0 +1,161 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package updateutils
+
+import (
+	"github.com/stretchr/testify/require"
+	commandpb "go.temporal.io/api/command/v1"
+	enumspb "go.temporal.io/api/enums/v1"
+	failurepb "go.temporal.io/api/failure/v1"
+	protocolpb "go.temporal.io/api/protocol/v1"
+	updatepb "go.temporal.io/api/update/v1"
+	"go.temporal.io/server/common/payloads"
+	"go.temporal.io/server/common/testing/protoutils"
+	"go.temporal.io/server/common/testing/testvars"
+)
+
+type (
+	UpdateUtils struct {
+		t require.TestingT
+	}
+
+	helper interface {
+		Helper()
+	}
+)
+
+func New(t require.TestingT) UpdateUtils {
```

Comment:
One of the roles of ObjectMother is to make tests more readable. Unfortunately, I am failing to understand what you are testing from reading the tests.

Reply:
These are not tests. These are helpers for tests. Tests are in `update_workflow_test.go` file. They require some understanding of other test helpers (which can be improved, I agree), but they are more or less understandable.

Reply:
I know, this is what is known as ObjectMother design pattern. Tests are further down and hard to understand. Extracting and renaming did not help.